### PR TITLE
Fix: unhandled error path when apiVersion or kind is undefined

### PR DESCRIPTION
**Pull request by alewitt2**
also handles when metadata.name is undefined

(fixes https://github.com/razee-io/MustacheTemplate/issues/144)

### CODE REVIEW

File: lib/CompositeController.js [hunk 1]
```diff
@@ -86,11 +108,37 @@ module.exports = class CompositeController extends BaseController {
...
-    let krm = await this.kubeClass.getKubeResourceMeta(child.apiVersion, child.kind, 'update');
+    if (!childName) {
+      return {
+        statusCode: 400,
+        body: {
+          kind: 'Status',
+          apiVersion: 'v1',
+          metadata: {},
+          status: 'Failure',
+          message: `Invalid kubernetes resource, 'metadata.name: ${childName}' must not be empty`,
```
Comment (adamkingit) [inline lib/CompositeController.js:119]: Won't ${childName} be undefined in this message?

File: lib/CompositeController.js [hunk 2]
```diff
@@ -71,7 +71,29 @@ module.exports = class CompositeController extends BaseController {
...
+          status: 'Failure',
+          message: `Invalid kubernetes resource, 'kind: ${childKind}' and 'apiVersion: ${childApiVersion}' must not be empty`,
+          reason: 'BadRequest',
+          details: { apiVersion: `${childApiVersion}`, kind: `${childKind}`, uri: childUri },
+          code: 400
+        }
+      };
+    }
+
+    if (childApiVersion.toLowerCase() == 'v1' && childKind.toLowerCase() == 'list' && Array.isArray(child.items)) {
```
Comment (adamkingit) [inline lib/CompositeController.js:96]:
```suggestion
    if (childApiVersion.toLowerCase() === 'v1' && childKind.toLowerCase() === 'list' && Array.isArray(child.items)) {
```

File: lib/CompositeController.js [hunk 3]
```diff
@@ -101,7 +149,7 @@ module.exports = class CompositeController extends BaseController {
     if (!objectPath.has(child, ['metadata', 'namespace']) && krm.namespaced) {
       objectPath.set(child, ['metadata', 'namespace'], this.namespace);
     }
-    let childUri = krm.uri({ name: child.metadata.name, namespace: child.metadata.namespace });
+    childUri = krm.uri({ name: childName, namespace: objectPath.get(child, 'metadata.namespace') });
```
Comment (adamkingit) [inline lib/CompositeController.js:152]:
```suggestion
    childUri = krm.uri({ name: childName, namespace: objectPath.get(child, 'metadata.namespace','default') });
```

Comment (alewitt2) [reply lib/CompositeController.js:152]: no i want this to be undefined if it is truely undefined

File: lib/CompositeController.js (hunk 1 above)
Comment (alewitt2) [reply lib/CompositeController.js:119]: yes. that was kind of the point. but for debuging future problems, i like to print it out just in case

**Review by adamkingit — APPROVED** [review]

**Pull request merged**